yourgov/components/FormMobileFoodVendorPermit/task1/FormTask1Step5.tsx

```diff
@@ -0,0 +1,148 @@
+import { Controller, SubmitHandler, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { DateRangePicker } from '@ag.ds-next/react/date-range-picker';
+import { FormStack } from '@ag.ds-next/react/form-stack';
+import { GroupedFields } from '@ag.ds-next/react/grouped-fields';
+import { Stack } from '@ag.ds-next/react/stack';
+import { TimeInput } from '@ag.ds-next/react/time-input';
+import { DeepPartial } from '../../../lib/types';
+import { FormPageAlert } from '../FormPageAlert';
+import { hasMultipleErrors, parseDateField } from '../utils';
+import { type ShallowErrors } from '../FormState';
+import { StepActions } from '../StepActions';
+import { useGlobalForm } from '../GlobalFormProvider';
+import { FormTask1Container } from './FormTask1Container';
+import { useFormTask1Context } from './FormTask1Provider';
+import {
+	task1Step5FormSchema,
+	type Task1Step5FormSchema,
+} from './FormTask1FormState';
+
+function transformDefaultValues(step?: DeepPartial<Task1Step5FormSchema>) {
+	const from = step?.tradingPeriod?.from as string | Date | undefined;
+	const to = step?.tradingPeriod?.from as string | Date | undefined;
+	return {
+		...step,
+		tradingPeriod: {
+			from: parseDateField(from),
+			to: parseDateField(to),
+		},
+	};
+}
+
+export function FormTask1Step5() {
+	const { formState, setFormState, isSavingBeforeExiting } = useGlobalForm();
+	const { submitStep } = useFormTask1Context();
+
+	const {
+		control,
+		handleSubmit,
+		formState: { errors },
+	} = useForm<Task1Step5FormSchema>({
+		defaultValues: transformDefaultValues(formState.task1?.step5),
+		resolver: isSavingBeforeExiting
+			? undefined
+			: zodResolver(task1Step5FormSchema),
+		mode: 'onSubmit',
+		reValidateMode: 'onBlur',
+	});
+
+	const typeCorrectedErrors = errors as ShallowErrors<Task1Step5FormSchema>;
+
+	const onSubmit: SubmitHandler<Task1Step5FormSchema> = async (data) => {
+		if (isSavingBeforeExiting) {
+			return;
+		}
+		await submitStep();
+		setFormState({
+			...formState,
+			task1: {
+				...formState.task1,
+				step5: {
+					...data,
+					completed: !isSavingBeforeExiting,
```

Comment:
```suggestion
					completed: true,
```

Reply:
Yeah, we need to figure out what these buttons do, because right now, neither is correct https://github.com/agriculturegovau/agds-next/pull/1734#discussion_r1714717373